src/util/EmojiImporter.cpp

```diff
@@ -0,0 +1,126 @@
+#include "EmojiImporter.h"
+
+EmojiImporter::EmojiImporter(QString outputDirectoryPath, QList<QUrl> files) {
+    this->outputDirectoryPath = outputDirectoryPath;
+
+    foreach (auto fileUrl, files) {
+        auto fileName = fileUrl.fileName().toLower();
+
+        if (!fileUrl.isLocalFile()) {
+            filesToDownload.append(fileUrl.toString());
+            continue;
+        }
+
+        if (fileName.endsWith(".zip")) {
+            filesToExtract.append(fileUrl.path());
+            continue;
+        }
+
+        for (const auto &ext : Config::getImageExtensions()) {
```

Comment:
this can be combined with the previous if statement by creating a local variable of image extensiosn and adding .zip to it.